### ISSUE TITLE
Tests fail (with Protocol::WebSocket 0.22?)

### ISSUE THREAD

**Issue by eserte** [plicease/AnyEvent-WebSocket-Client#33]
Tests started to fail on my smokers:
```
        # Failed test at t/anyevent_websocket_connection.t line 154.
        # Unicode::GCString is not installed, table may not display all unicode characters properly
        # +---------------------+---------------------+----+----------+----------+
        # | PATH                | GOT                 | OP | CHECK    | LNs      |
        # +---------------------+---------------------+----+----------+----------+
        # |                     | AnyEvent::WebSocket |    | <OBJECT> | 150, 153 |
        # |                     | ::Connection=HASH(0 |    |          |          |
        # |                     | x44ca5cf0)          |    |          |          |
        # |                     |                     |    |          |          |
        # | close_code()        | 963                 | eq | 1005     | 440      |
        # | close_reason()      | ï¿½bb                 | eq | bb       | 440      |
        # +---------------------+---------------------+----+----------+----------+
    # Failed test 'close'
    # at t/anyevent_websocket_connection.t line 155.

(etc)
```

Statistical analysis suggests that the problem started with Protocol::WebSocket 0.22 (@vti @leonerd: FYI):
```
****************************************************************
Regression 'mod:Protocol::WebSocket'
****************************************************************
Name           	       Theta	      StdErr	 T-stat
[0='const']    	      1.0000	      0.0000	63928868152459800.00
[1='eq_0.21']  	      0.0000	      0.0000	   2.74
[2='eq_0.22']  	     -1.0000	      0.0000	-40146423047850984.00

R^2= 1.000, N= 135, K= 3
****************************************************************
```

**Comment by plicease** [comment]
Indeed, these changes are likely the trigger:
https://metacpan.org/diff/file?target=VTI/Protocol-WebSocket-0.22/lib%2FProtocol%2FWebSocket%2FFrame.pm&source=VTI%2FProtocol-WebSocket-0.21%2Flib%2FProtocol%2FWebSocket%2FFrame.pm

**Comment by vti** [comment]
Sorry, blame @leonerd for that :)

**Comment by leonerd** [comment]
Oops. This is almost certainly related to Unicode string vs. UTF-8 character handling in the surrounding code.

If someone can give me a bit of hand-holding about development/setup/testing on this code I'll be happy to take a look and see if I can fix it.

**Comment by leonerd** [comment]
Oh actually it's a ~2line patch to fix.

[GH33.patch.txt](https://github.com/plicease/AnyEvent-WebSocket-Client/files/1552712/GH33.patch.txt)

**Comment by plicease** [comment]
Patch seems to work with `0.20`, `0.21`, `0.22` and `0.23`.  Since the change was reverted in `0.23` it isn't necessary there (?), but it doesn't seem to do any harm either.

**Comment by vti** [comment]
It effected other modules that's why I reverted it for now.

**Comment by plicease** [comment]
yeah. I gathered.

I've released this fix as WSC `0.44`, as it fixes compat with PW `0.22` and doesn't seem to break any of the other versions that we support.  Please let me know if there are any changes in WSC required for good fix referred to here: https://github.com/vti/protocol-websocket/issues/54#issuecomment-351162385

**Comment by vti** [comment]
I will try to make sure dependent modules keep working %)
